wireapp/wire-ios-mono
# fix: call participant not removed from subconversation WPB-3061

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-3061" title="WPB-3061" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-3061</a>  [iOS] Participant still in MLS conference after leaving
  </td></table>
  <br />
 






----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Given I (user A) am in an MLS conference with another user B, when B hangs up to leave the call, I still see B in my call participant list.

### Causes

When B leaves the call, the backend will send a remove proposal on their behalf to all other participants. A receives the proposal, but when they try to commit pending proposals Core Crypto returns an error stating there exist no pending proposals. Therefore B remains in the subconversation and the conference.

The reason why Core Crypto thinks there are no pending proposals is because we actually to try commit pending proposals in the parent group when the proposal is actually for the subgroup.

### Solutions

When committing pending proposals, check if a conversation has any sub conversations. If it doesn't, include those as part of the list of groups containing possible pending proposals.

### Testing

#### Test Coverage 

- Added a unit test asserting we commit pending proposals for subgroups if they exist

#### How to Test

- Be in an an MLS conference with user B.
- User B hangs up.
- Check for user B they left the call.
- Check for user A they left the call.

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
